### PR TITLE
validate and update vendor targets

### PR DESCRIPTION
**Pull request by crazy-max**
follow-up #3315

discussed with @milosgajdos about cleaning up our workflows that are a bit redundant (like the `ci.yml` one).

this pr adds targets to update and validate vendor:

```
$ make vendor
$ make validate-vendor
```

like our build workflow, everything is sandboxed.

another target has been added to check for outdated dependencies:

```
$ make mod-outdated
...
#15 10.41 +-----------------------------------------------+------------------------------------+------------------------------------+--------+------------------+
#15 10.41 |                    MODULE                     |              VERSION               |            NEW VERSION             | DIRECT | VALID TIMESTAMPS |
#15 10.41 +-----------------------------------------------+------------------------------------+------------------------------------+--------+------------------+
#15 10.41 | github.com/Azure/azure-sdk-for-go             | v56.3.0+incompatible               | v63.3.0+incompatible               | true   | true             |
#15 10.41 | github.com/aws/aws-sdk-go                     | v1.43.16                           | v1.43.44                           | true   | true             |
#15 10.41 | github.com/bshuster-repo/logrus-logstash-hook | v1.0.0                             | v1.0.2                             | true   | true             |
#15 10.41 | github.com/bugsnag/bugsnag-go                 | v0.0.0-20141110184014-b1d153021fcd | v2.1.2+incompatible                | true   | true             |
#15 10.41 | github.com/denverdino/aliyungo                | v0.0.0-20190125010748-a747050bb1ba | v0.0.0-20220419071826-816031062a6e | true   | true             |
#15 10.41 | github.com/docker/libtrust                    | v0.0.0-20150114040149-fa567046d9b1 | v0.0.0-20160708172513-aabc10ec26b7 | true   | true             |
#15 10.41 | github.com/gomodule/redigo                    | v1.8.2                             | v1.8.8                             | true   | true             |
#15 10.41 | github.com/mitchellh/mapstructure             | v1.1.2                             | v1.5.0                             | true   | true             |
#15 10.41 | github.com/ncw/swift                          | v1.0.47                            | v1.0.53                            | true   | true             |
#15 10.41 | github.com/spf13/cobra                        | v1.0.0                             | v1.4.0                             | true   | true             |
#15 10.41 | github.com/yvasiyarov/gorelic                 | v0.0.0-20141212073537-a9bba5b9ab50 | v0.0.7                             | true   | true             |
#15 10.41 | golang.org/x/crypto                           | v0.0.0-20211215153901-e495a2d5b3d3 | v0.0.0-20220411220226-7b82a4e95df4 | true   | true             |
#15 10.41 | golang.org/x/oauth2                           | v0.0.0-20190604053449-0f29369cfe45 | v0.0.0-20220411215720-9780585627b5 | true   | true             |
#15 10.41 | google.golang.org/api                         | v0.0.0-20160322025152-9bf6e6e569ff | v0.75.0                            | true   | true             |
#15 10.41 | google.golang.org/cloud                       | v0.0.0-20151119220103-975617b05ea8 | v0.101.0                           | true   | true             |
#15 10.41 | gopkg.in/check.v1                             | v1.0.0-20180628173108-788fd7840127 | v1.0.0-20201130134442-10cb98267c6c | true   | true             |
#15 10.41 +-----------------------------------------------+------------------------------------+------------------------------------+--------+------------------+
#15 DONE 10.4s
```

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -47,7 +47,6 @@ jobs:
       run: |
         DCO_VERBOSITY=-q script/validate/dco
         GO111MODULE=on script/setup/install-dev-tools
```
Comment (milosgajdos) [inline .github/workflows/ci.yml:49]: I'm wondering, do we still need `GO111MODULE`? 🤔

Comment (crazy-max) [reply .github/workflows/ci.yml:49]: Yeah in a follow-up we should remove the ci workflow. In a follow-up will take a look at the DCO check but DCO bot might be enough?

Comment (milosgajdos) [reply .github/workflows/ci.yml:49]: Yeah, I think the DCO bot should take care of that...

File: .github/workflows/validate.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      - '*'
+
+jobs:
+  validate:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        target:
+          - validate-vendor
```
Comment (milosgajdos) [inline .github/workflows/validate.yml:21]: Does this somehow automagically trigger `make validate-vendor`?

Comment (crazy-max) [reply .github/workflows/validate.yml:21]: Yes exactly. The idea in the follow-up is to add other validators to this matrix like `lint`, `authors`, etc... See https://github.com/docker/buildx/blob/a6a1a362ad951af488fc41388f41a2d52b25f791/.github/workflows/validate.yml#L23-L25

Comment (milosgajdos) [reply .github/workflows/validate.yml:21]: Oh neat!

**Review by milosgajdos — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (wy65701436) [reply .github/workflows/ci.yml:49]: +1 for remove `GO111MODULE`

**Review by wy65701436 — APPROVED** [review]
lgtm

**Pull request merged**